### DISCUSSION TITLE
How to add a validationResolver but keep existing rules ?

### DISCUSSION THREAD

**Discussion by maximelebastard** [react-hook-form/react-hook-form#2022] [Q&A] [answered]
Hi,

I have a lot of forms made of Controllers with simple rules (usually only `{required: true}`).
Recently, I needed a slightly more advanced validation (check that a date is after another), so I added a validationResolver that uses Joi.

The date check works well, but I get no errors anymore about the required fields.

How can I make the validationResolver being only some "additional" validation ?

Thanks
Max

My validationResolver

```
const validationResolver: ValidationResolver<ModalDataOut> = (data) => {
  const { error, value: values } = validationSchema.validate(data, { abortEarly: false });
  const aa = {
    values: error ? {} : values,
    errors: error
      ? error.details.reduce((previous, currentError) => {
          return {
            ...previous,
            [currentError.path[0]]: currentError
          };
        }, {})
      : {}
  };

  console.log({ aa });
  return aa;
};
```

One of my controlled inputs on the form. That input doesn't trigger any error anymore if it is empty.
```
            <InputLocation
              name="location"
              label="Adresse"
              withEstates={true}
              disabled={modalContext.disabled}
              placeholder="Saisissez une adresse"
              control={control}
              rules={{ required: true }}
              error={errors.location}
            />
```

**Comment by bluebill1049** [comment, marked as answer]
you can't use both, you have to pick either.

**Reply by maximelebastard** [reply]
Thanks for the quick reply Bill 👍  Let's refactor then

**Comment by AgathiaKimiya** [comment]
Hello! I had something similar where I needed to add rules to some inputs (regex validation in addition to required or something else) and stumble upon this thread and did not find much answer.

I did some code that makes it possible, not the cleanest one, it's not possible to chain calls to the method that adds a rule (though some refactoring would make it possible), but it works in all the cases I could think of :
- No rules at all
- Required or something else that is not "validate"
- One validate
- Several rules in validate

Here's the snippet, if it helps anyone stumbling here :
https://codesandbox.io/s/elastic-voice-rie3r?file=/index.js
